### ISSUE TITLE
Does the WANN method work on toy regression data？

### ISSUE THREAD

**Issue by jackstramp** [adapt-python/adapt#95]
Does the WANN method work on toy regression data（make_regression_da）? Can you give an example? I tried it myself, it didn't work well

**Comment by antoinedemathelin** [comment]
Hi @jackstramp,
Thank you for your interest in the ADAPT library!

Here is an example for WANN on make_regression_da. Indeed, WANN is a supervised DA method which uses a small batch of labeled data from the target domain. To make WANN works ("well") you need to pass the target labels in both source and target entries:

```python
from adapt.utils import make_regression_da
from adapt.instance_based import WANN

Xs, ys, Xt, yt = make_regression_da()
model = WANN(Xt=Xt[:3], yt=yt[:3], random_state=0)
model.fit(np.concatenate((Xs, Xt[:3])),
          np.concatenate((ys, yt[:3])), epochs=600, verbose=0)
model.score(Xt, yt)
```

Then to visualize the results:

```python
import numpy as np
import matplotlib.pyplot as plt

x = np.linspace(-1, 1, 100)
yp = model.predict(x.reshape(-1, 1))

plt.scatter(Xs, ys, s=model.predict_weights(Xs).ravel()**3, label="weighted source")
plt.plot(Xt, yt, ".", c="C1", label="target unlabeled")
plt.plot(Xt[:3], yt[:3], ".", c="k", label="target labeled")
plt.plot(x, yp, c="r", "predictions")
```

You get this figure:
![image](https://user-images.githubusercontent.com/45264923/224265714-fcae955d-96b2-49db-8609-6200c9bb0d92.png)


Note that make_regression_da is a counter-example for most instance-based DA, here WANN is behaving correctly by increasing the weights of source data close to the target but, in this particular example, the optimal weighting would do the opposite. In fact, this toy problem has been tailored for [TrAdaBoostR2](https://adapt-python.github.io/adapt/generated/adapt.instance_based.TrAdaBoostR2.html) which works well in this case.

**Comment by jackstramp** [comment]
Thank you for your reply, I still have two questions,
1,  for “model = WANN(Xt=Xt[:3], yt=yt[:3], random_state=0)“, the target labels have passed to model, so  in the fitting stage, Is the concatenate operation necessary? Can we only pass the source labels to the fit function ? 
2.  About "make_regression_da is a counter-example for most instance-based DA" and "the optimal weighting would do the opposite", Can you explain more in detail？
Thank you very much again

**Comment by antoinedemathelin** [comment]
For question 1:
Yes, indeed, the target labels are already given to the model in "WANN(Xt=Xt[:3], yt=yt[:3])", it would be better to make the concatenation with the source inside the WANN `fit` function. We will correct this in a future version.

For question 2:
The general assumption made by instance-based method is that: "two close data points in the input space "X" should have close labels "y" independently from the fact that the data points comes form the source or the target domain" (this example is suited for instance-based methods: [sample 1d](https://adapt-python.github.io/adapt/examples/sample_bias.html)). 
In the make_regression_da example the source and target data points close to zeros have very different y values which is not in line with the general instance-based DA assumption.

**Comment by jackstramp** [comment]
Thanks a lot
